packages/yarnpkg-core/sources/miscUtils.ts

```diff
@@ -261,6 +264,23 @@ export function dynamicRequireNoCache(path: PortablePath) {
   return result;
 }
 
+const dynamicRequireFreshCache = new Map<PortablePath, { mtime: number, instance: any }>();
+
+/**
+ * Requires a module without using the cache if it has changed since the last time it was loaded
+ */
+export function dynamicRequireFresh(path: PortablePath) {
```

Comment:
I read "fresh" as implying the require is always up-to-date. Perhaps instead the public API could be:

```ts
type RequireCache = 'mtime' | null;

export function dynamicRequire(path: PortablePath, opts: {cache: RequireCache}) {
```

Also, perhaps `dynamicRequireNoCache` should burst the entry in `dynamicRequireFreshCache`?